lib/spack/spack/build_systems/python.py

```diff
@@ -178,6 +181,37 @@ def install(self, spec, prefix):
         with working_dir(self.build_directory):
             pip(*args)
 
+    @property
+    def headers(self):
+        """Discover header files in platlib."""
+
+        # Headers may be in either location
+        include = inspect.getmodule(self).include
+        platlib = inspect.getmodule(self).platlib
+        headers = find_all_headers(include) + find_all_headers(platlib)
+
+        if headers:
+            return headers
+
+        msg = 'Unable to locate {} headers in {}'
+        raise NoHeadersError(msg.format(self.spec.name, root))
+
+    @property
+    def libs(self):
+        """Discover libraries in platlib."""
+
+        # Remove py- prefix in package name
+        library = 'lib' + self.spec.name[3:].replace('-', '?')
+        root = inspect.getmodule(self).platlib
+
+        for shared in [True, False]:
```

Comment:
(minor, optional) it's probably rare for `py-*` packages to specify `~shared`, but do you think it might be worth checking for that variant? IMO even if so, it can be done another time.